types/hapi/index.d.ts

```diff
@@ -18,54 +18,3869 @@
+
+/**
+ * Access: read only and the public podium interface.
+ * The response.events object supports the following events:
+ * * 'peek' - emitted for each chunk of data written back to the client connection. The event method signature is function(chunk, encoding).
+ * * 'finish' - emitted when the response finished writing but before the client response connection is ended. The event method signature is function ().
+ * [See docs](https://hapijs.com/api/17.0.1#-responseevents)
+ */
+export interface ResponseEvents extends Podium {
+    /**
+     * 'peek' - emitted for each chunk of data written back to the client connection. The event method signature is function(chunk, encoding).
+     * 'finish' - emitted when the response finished writing but before the client response connection is ended. The event method signature is function ().
+     */
+    on(criteria: 'peek', listener: PeekListener): void;
+
+    on(criteria: 'finish', listener: (data: undefined) => void): void;
+
+    /**
+     * 'peek' - emitted for each chunk of data written back to the client connection. The event method signature is function(chunk, encoding).
+     * 'finish' - emitted when the response finished writing but before the client response connection is ended. The event method signature is function ().
+     */
+    once(criteria: 'peek', listener: PeekListener): void;
+
+    once(criteria: 'finish', listener: (data: undefined) => void): void;
+}
+
+/**
+ * Object where:
+ *  * append - if true, the value is appended to any existing header value using separator. Defaults to false.
+ *  * separator - string used as separator when appending to an existing value. Defaults to ','.
+ *  * override - if false, the header value is not set if an existing value present. Defaults to true.
+ *  * duplicate - if false, the header value is not modified if the provided value is already included. Does not apply when append is false or if the name is 'set-cookie'. Defaults to true.
+ * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-responseheadername-value-options)
+ * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#response-object)
+ */
+export interface ResponseObjectHeaderOptions {
+    append?: boolean;
+    separator?: string;
+    override?: boolean;
+    duplicate?: boolean;
+}
+
+/**
+ * The response object contains the request response value along with various HTTP headers and flags. When a lifecycle
+ * method returns a value, the value is wrapped in a response object along with some default flags (e.g. 200 status
+ * code). In order to customize a response before it is returned, the h.response() method is provided.
+ * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#response-object)
+ * TODO, check extending from Podium is correct.  Extending because of "The response object supports the following events" [See docs](https://hapijs.com/api/17.0.1#-responseevents)
+ */
+export interface ResponseObject extends Podium {
```

Comment:
@SimonSchick I just commented on the wrong line, it still needs to be done.

Reply:
I overlooked this, but it will fail to compile as soon as I included Boom.BoomError.  For instance:

```
export interface ResponseObject extends Podium, Boom.BoomError {
```

However, there's a type conflict with `response.message`.  ResponseObject defines it as a function.  Boom defines `message` as a string.  The only way it will compile is if I change it to `message: any`, which I don't really want to do.  Is there another way of handling this?

Reply:
No, the `response` property on request is `RequestObject | Boom.BoomError`.